executor/src/witgen/jit/prover_function_heuristics.rs

```diff
@@ -59,33 +72,41 @@ fn decode_prover_function<T: FieldElement>(
     function: &Expression,
     try_symbol_by_name: impl TrySymbolByName,
 ) -> Result<ProverFunction<'_, T>, String> {
-    if let Some(provide_if_unknown) =
-        try_decode_provide_if_unknown(index, function, try_symbol_by_name)
-    {
-        Ok(provide_if_unknown)
+    let results = [
+        try_decode_provide_if_unknown(index, function, try_symbol_by_name),
```

Comment:
very minor, but this way its always evaluating all three, instead of stopping on the first success... if the functions have the same signature, maybe its possible to do something like `[try_func1, try_func2, try_func3].iter().map(|f| f(...))`?

Reply:
It will fail rather quickly and it makes it easy to access all the errors.